app/controllers/application_controller.rb

```diff
@@ -22,12 +23,11 @@ def append_to_log(params)
   # Rails' instrumentation code, and is run after each request.
   def append_info_to_payload(payload)
     super
-    if @custom_log_items
-      payload[:custom_log_items] = @custom_log_items
-    end
-  end
 
-  # :nocov:
+    append_to_log(request_id: RequestStore.store[:request_id])
```

Comment:
Because it creates more indirection, which I find worse.

To me hooking it up directly to the method that Rails provides for log instrumentation is the cleanest solution instead of adding indirection to achieve the same result. Basically trying to avoid to make even more confusing.

I also don't particularly like before actions because is not simple to reason about them when there are many of them as they create indirection when reading code, so if we can avoid some of them better.